chat/client-api.md

```diff
@@ -0,0 +1,62 @@
+# Client API
+
+```typescript
+abstract class Client {
+  // ---------- Methods ----------------------------------------------- //
+
+  // initializes the client with persisted storage and a network connection
+  public abstract init(): Promise<void>;
+
+  // register a blockchain account with a public key / returns the public key
+  public abstract register(params: {
+    account: string;
+  }): Promise<string>;
+
+  // queries the default keyserver with a blockchain account / returns the public key
+  public abstract resolve(params: {
+    account: string;
+  }): Promise<string>;
+
+  // sends a chat invite with opening message / returns an invite id
+  public abstract invite(params: {
+    peerPublicKey: string;
+    invite: Invite;
+  }): Promise<number>;
+
+  // accepts a chat invite by id / returns thread topic
+  public abstract accept(params: {
+    inviteId: string;
+  }): Promise<string>;
+
```

Comment:
```suggestion

  public abstract getMessages(params: {
    threadId: string;
  }): Promise<[Messages]>;
  
 public abstract getThreads(): Promise<[Threads]>;
  
```

make sense?

Reply:
Agreed, we currently don't specify here how the relevant entities (messages, threads, invites) are resolved. I guess it's implied this would happen in `init` right now and then the listeners (e.g. `on("chat_message,...)`) would take it from there?

So additionally `public abstract getInvites()` to cover all handled data structures?